environment/src/main/java/com/sequenceiq/environment/environment/sync/EnvironmentSyncService.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.environment.environment.sync;
+
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.environment.environment.EnvironmentStatus;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.service.freeipa.FreeIpaService;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.describe.DescribeFreeIpaResponse;
+
+@Component
+public class EnvironmentSyncService {
+
+    private final FreeIpaService freeIpaService;
+
+    public EnvironmentSyncService(FreeIpaService freeIpaService) {
+        this.freeIpaService = freeIpaService;
+    }
+
+    public EnvironmentStatus getStatusByFreeipa(Environment environment) {
+        Optional<DescribeFreeIpaResponse> freeIpaResponseOpt = freeIpaService.describe(environment.getResourceCrn());
+        if (freeIpaResponseOpt.isPresent()) {
+            DescribeFreeIpaResponse freeIpaResponse = freeIpaResponseOpt.get();
+            if (freeIpaResponse.getStatus() == Status.STOPPED) {
+                return EnvironmentStatus.ENV_STOPPED;
+            } else if (freeIpaResponse.getStatus() == Status.DELETED_ON_PROVIDER_SIDE) {
+                return EnvironmentStatus.FREEIPA_DELETED_ON_PROVIDER_SIDE;
+            } else if (freeIpaResponse.getStatus() == Status.STOP_FAILED) {
+                return EnvironmentStatus.STOP_FREEIPA_FAILED;
+            } else if (freeIpaResponse.getStatus() == Status.START_FAILED) {
+                return EnvironmentStatus.START_FREEIPA_FAILED;
+            } else if (freeIpaResponse.getStatus() == Status.START_IN_PROGRESS) {
+                return EnvironmentStatus.START_FREEIPA_STARTED;
+            } else if (freeIpaResponse.getStatus() == Status.STOP_IN_PROGRESS) {
+                return EnvironmentStatus.STOP_FREEIPA_STARTED;
+            }
```

Comment:
switch?

Reply:
I do not like `switch` because of the `checkstyle`. Everytime crying about "cyclomaticComplexity", but I cannot simplify the case number. But I make a try with the `switch`